pymongo/topology.py

```diff
@@ -430,21 +475,22 @@ def _get_replica_set_members(self, selector):
             ):
                 return set()
 
-            return {sd.address for sd in selector(self._new_selection())}
+            # mypy doesn't recognize classes that implement a __getitem__ as something that is iterable
+            return {sd.address for sd in selector(self._new_selection())}  # type: ignore[attr-defined]
```

Comment:
Does calling iter() explicitly fix this: `iter(selector(...))`?

Reply:
https://github.com/python/mypy/issues/2220

Reply:
yes! good idea!